src/initialize/init.cpp

```diff
@@ -50,8 +46,7 @@ void finalize() {
 }
 
 initializer::initializer() {
-    initialize();
-    lean::io_mark_end_initialization();
+    lean_initialize();
```

Comment:
Could you please add a comment here saying that we used to have `lean::io_mark_end_initialization` here and why we moved it to another place?
It is just a matter of copying the explanation above as a comment here.
We should also add a reminder that people using Lean as a library in their code are now responsible for invoking `lean::io_mark_end_initialization` themselves.